src/Microsoft.ApplicationInsights.AspNet/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -53,41 +54,16 @@ public static void AddApplicationInsightsTelemetry(this IServiceCollection servi
                 return telemetryConfiguration;
             });
 
+            services.AddSingleton<IJavaScriptSnippet, ApplicationInsightsJavaScript>();
+
             services.AddScoped<TelemetryClient>();
 
             services.AddScoped<RequestTelemetry>((svcs) => {
```

Comment:
Yes, it tries to call not-default constructor and fails otherwise. I'll add a comment